### PR TITLE
Fix raw tag

### PR DESCRIPTION
**Pull request by Zhao-Andy**
## What type of PR is this? (check all applicable)
- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Documentation Update

## Description
Fixes the raw tag and prevents it from being used incorrectly.

## Added to documentation?
- [x] no documentation needed

### CODE REVIEW

File: app/liquid_tags/raw_tag.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module Liquid
+  class Raw < Block
+    FullTokenPossiblyInvalid = /\A(.*)#{TagStart}\s*(\w+)\s*#{TagEnd}\z/om
```
Comment (Zhao-Andy) [inline app/liquid_tags/raw_tag.rb]: Renaming the constant like this prints the following warning:
```
dev.to/app/liquid_tags/raw_tag.rb:3: warning: already initialized constant Liquid::Raw::FullTokenPossiblyInvalid
.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/liquid-4.0.1/lib/liquid/tags/raw.rb:4: warning: previous definition of FullTokenPossiblyInvalid was here
```

Not sure how to get around it, but I think this warning will only at load time.

Comment (maestromac) [reply app/liquid_tags/raw_tag.rb]: I can't find it right now but there's a sensible way to reassign class's constant so that warning appear everywhere.

**Comment by benhalpern** [comment]
Looks like the failing test may be random and unrelated to this. Any ideas on that front?

File: spec/features/user_edits_an_article_spec.rb [hunk 2]
```diff
@@ -29,6 +29,9 @@
   end
 
   it "user update their post" do
+    if article
+      10.times { puts "article exists" }
+    end
     visit "/#{user.username}/#{article.slug}/edit"
     fill_in "article_body_markdown", with: template.gsub("true", "false")
```
Comment (benhalpern) [inline spec/features/user_edits_an_article_spec.rb]: Hmmm "article_body_markdown" is an element that only exists if the user is logged in and has permission to visit the article _and_ has "v1" editor as their editor. Do we know whether they meet all these criteria?

File: app/liquid_tags/raw_tag.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+module Liquid
+  class Raw < Block
+    remove_const(:FullTokenPossiblyInvalid) if defined?(FullTokenPossiblyInvalid)
```
[the remark this answers is not in the document]
Comment (benhalpern) [reply app/liquid_tags/raw_tag.rb]: Should be caught in tests if it becomes a problem.

**Pull request merged**